sherpa/data.py

```diff
@@ -487,14 +492,45 @@ def apply(self, array):
             raise DataErr('mismatch', 'mask', 'data array')
         return array[self.mask]
 
-    def notice(self, mins, maxes, axislist, ignore=False):
+    def notice(self, mins, maxes, axislist, ignore=False, integrated=False):
+        """Select a range.
+
+        Parameters
+        ----------
+        mins : sequence of values
+           The minimum value of the valid range (elements may be
+           None).  When not None, it is treated as an inclusive limit,
+           so points >= min are included.
+        maxes : sequence of values
+           The maximum value of the valid range (elements may be
+           None).  When not None, it is treated as an inclusive limit,
+           so points <= max are included, when upper is True,
```

Comment:
Where is upper set/accessed?

Reply:
Just to note that I didn't design this code - I'm just trying to document it. 

Saying that,  I don't understand the question.

Reply:
Reading the test 'when upper is True' sounds like upper is a parameter to the function (which it doesn't appear to be) so I was asking for clarification of what determines when 'points <= max are included'.

Reply:
Aha. This is botched documentation on my part - I think early on I had used a parameter called `upper` but I've changed this - so now I think it is likely related to `integrated` flag but I'll have to review things as the semantics of the two parameters are probably different.

Reply:
Addressed in https://github.com/sherpa/sherpa/pull/1216/commits/73ca7cc38e045b2883205f5a11b497a90e8a49bf